lib/flagutil/map.go

```diff
@@ -0,0 +1,37 @@
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+)
+
+type MapString map[string]string
+
+func (m *MapString) String() string {
+	return fmt.Sprintf("%v", *m)
+}
+
+func (m *MapString) Set(value string) error {
+	if *m == nil {
+		*m = make(map[string]string)
+	}
+	for _, pair := range parseArrayValues(value) {
+		key, value := parseMapValue(pair)
+		(*m)[key] = value
+	}
+	return nil
+}
+
+func parseMapValue(s string) (string, string) {
+	kv := strings.SplitN(s, ":", 2)
+	return kv[0], kv[1]
```

Comment:
this panic if no `:` see https://go.dev/play/p/9aA8Dk9qHg1
which is not bad but we can panic more explicitly

Reply:
True ideally one would present an error to the user that makes sense. Is there are preferred way to do that here?

Reply:
Actually just spotted I can through an error all the way back up the stack which is nicer.